model/inline/src/main/scala/aqua/model/inline/MakeStructRawInliner.scala

```diff
@@ -6,29 +6,49 @@ import aqua.model.inline.state.{Arrows, Exports, Mangler}
 import aqua.model.*
 import aqua.raw.value.MakeStructRaw
 import aqua.types.{StreamMapType, StructType}
+
 import cats.data.{Chain, NonEmptyMap, State}
 import cats.syntax.foldable.*
+import cats.syntax.bifunctor.*
 import cats.syntax.functor.*
 
 object MakeStructRawInliner extends RawInliner[MakeStructRaw] {
 
+  // return unfolding ops and generated insertions
+  def genInsertion[S: Mangler](
```

Comment:
Lets give this function a better name and full doc?
I suggest smth like `constructThroughMap`